### ISSUE TITLE
Crashes when non-axisymmetric (N>0) boundary surface is used

### ISSUE THREAD

**Issue by patil3** [ddudt/DESC#82]
The code crashes whenever using non-zero N.

### Configuration/Environment
Currently Loaded Modulefiles:
1) gmp-6.2.0    
2) mpfr-4.1.0
3) mpc-1.1.0    
4) isl-0.22.1
5) gcc-10.2.0/gcc-10.2.0
6) python-3.9.5 

Operating System: Ubuntu 18.04.5 LTS
Kernel: Linux 4.15.0-112-generic

DESC version 0.4.3, using JAX backend, jax version=0.2.10, jaxlib version=0.1.62, dtype=float64
Using device: CPU, with 113.99 GB available memory

### Steps to reproduce the issue
There are a few different ways to reproduce the issue, all linked to using non-zero N.

**1. Following the "Using DESC interactively" example in documentation**
The example was followed line-for-line. This gave an error when the solution for a non-axisymmetric equilibrium was attempted using `eq2 = eq1.perturb(dRb=delta_R, dZb=delta_Z,order=2)`. The error occurs when the compute function is called in the code (full error message in attached file):
```
File "/home/spatil/.local/lib/python3.9/site-packages/desc/perturbations.py", line 114, in perturb
    RHS1 = eq.objective.compute(*args)
  File "/home/spatil/.local/lib/python3.9/site-packages/jaxlib/pocketfft.py", line 93, in pocketfft
    pocketfft_shape = builder.EndVector(n)
jax._src.traceback_util.FilteredStackTrace: TypeError: EndVector() takes 1 positional argument but 2 were given
```
All the python commands leading up to this are attached. The error that pops up is also attached.
[interactive1_script.txt](https://github.com/ddudt/DESC/files/6553138/interactive1_script.txt)
[interactive1_error.txt](https://github.com/ddudt/DESC/files/6553126/interactive1_error.txt)

**2. Starting with a boundary surface with N=1 in the interactive mode**
After the above issue, instead of perturbing the original surface, I tried starting with the non-axisymmetric boundary surface itself. The same error pops up. Commands and error attached as text.
[interactive2_script.txt](https://github.com/ddudt/DESC/files/6553154/interactive2_script.txt)
[interactive2_error.txt](https://github.com/ddudt/DESC/files/6553155/interactive2_error.txt)

**3. Using command line interface for the heliotron example cited in the paper**
I also tried using the example provided in the repository thinking that perhaps the issue was in the way I was inputting commands. `python3.9 -m desc -o out.output -pp -v ../DESC/examples/DESC/HELIOTRON >> disp.txt`. But again, the same error came up, leading me to believe that this may be a general issue. 
[disp.txt](https://github.com/ddudt/DESC/files/6553199/disp.txt)
[helio_cmdline_error.txt](https://github.com/ddudt/DESC/files/6553200/helio_cmdline_error.txt)

Since the example in the documentation clearly shows results, perhaps there is a problem with my configuration or some compatibility issue between all the dependencies (jaxlib, etc.).

**Comment by dpanici** [comment]
The latest commit (37eb8a35e42dd76c8cdfffcb957e1cacbb6278f3) on dev addresses/fixes this (it is an issue with flatbuffer and JAX), if you do `pip install flatbuffer==1.12`, this should fix the issue

**Comment by f0uriest** [comment]
Note the correct package is `flatbuffers==1.12` (plural). They also fixed the underlying issue in JAX starting with jaxlib v0.1.66 but we haven't fully tested some of their other changes (part of the fun of depending on a package still in unstable development) so we've stuck with an older version of jax for now. Expect updates in the near future.

**Comment by benjaminfaber** [comment]
Great, thank you guys.  I installed the dev branch and was able to run the `N=1` heliotron case from the documentation successfully.  One quick note, the default toroidal slices in `plot_surfaces` currently does not match what's on the website, but it sounds like that will be addressed in #83 when the notebook is updated.
